backend/packages/Upgrade/src/api/controllers/BatchAssignController.ts

```diff
@@ -0,0 +1,142 @@
+   *                required:
+   *                  - site
+   *                  - target
+   *                  - condition
+   *                properties:
+   *                  site:
+   *                    type: string
+   *                    minLength: 1
+   *                  target:
+   *                    type: string
+   *                    minLength: 1
+   *                  experimentType:
+   *                    type: string
+   *                    enum: [Simple, Factorial]
+   *                  assignedCondition:
+   *                    type: array
+   *                    items:
+   *                      type: object
+   *                      properties:
+   *                         conditionCode:
+   *                          type: string
+   *                          minLength: 1
+   *                         payload:
+   *                          type: object
+   *                          properties:
+   *                            type:
+   *                              type: string
+   *                            value:
+   *                              type: string
+   *                         id:
+   *                          type: string
+   *                         experimentId:
+   *                          type: string
+   *              example:
+   *                {
+   *                  "user1": {
+   *                    "site": "site1",
+   *                    "target": "target1",
+   *                    "condition": "condition1",
+   *                    "experimentType": "Simple",
+   *                    "assignedCondition": [
+   *                      {
+   *                        "conditionCode": "condition1",
+   *                        "payload": {
+   *                          "type": "type1",
+   *                          "value": "value1"
+   *                        },
+   *                        "id": "conditionId1",
+   *                        "experimentId": "experimentId1"
+   *                      }
+   *                    ]
+   *                  },
+   *                  "user2": null
+   *                }
+   *          '400':
+   *            description: BadRequestError - InvalidParameterValue
+   *          '401':
+   *            description: AuthorizationRequiredError
+   *          '500':
+   *            description: Internal Server Error
+   */
+  @Post('/')
+  public async getBatchAssignments(
+    @Body({ validate: true }) requestBody: BatchAssignValidator,
+    @Req() request: AppRequest
+  ): Promise<Record<string, IExperimentAssignmentv5 | null>> {
+    request.logger.info({ message: 'Request received for batch assignments' });
+    const { context, site, target, userIds } = requestBody;
+    request.logger.info({
+      message: `Context: ${context}, Site: ${site}, Target: ${target}, User IDs: ${userIds.join(', ')}`,
+    });
+
+    // Initialize response with nulls for all userIds
+    const nullResponse = userIds.reduce((acc, userId) => {
```

Comment:
I think we should return something different for users who are not found. Technically, we don't know what their assignment would be, not enough data to tell. It may be arguable if this matters for the experiments we are planning... but it could, and it could matter to someone using this endpoint later for something else.

Not sure what to suggest though, potentially we return `undefined` or leave that user out of the response, or return `NOT_FOUND` string instead of an assignment object?

Reply:
This would be a change in the spec. We can discuss on the board item.